Libraries/plist/Tools/PlistBuddy.cpp

```diff
@@ -206,6 +172,34 @@ CommandHelp()
 #undef INDENT
 }
 
+static std::unique_ptr<plist::Object>
+ReadPropertyList(Filesystem const *filesystem, std::string const &path)
+{
+    if (path.empty()) {
+        fprintf(stderr, "error: no input specified\n");
+        return NULL;
+    }
+
+    if (!filesystem->exists(path)) {
+        fprintf(stderr, "error: file does not exist: %s\n", path.c_str());
+        return NULL;
+    }
+
+    std::vector<uint8_t> contents;
+    if (!filesystem->read(&contents, path)) {
+        fprintf(stderr, "error: unable to read %s\n", path.c_str());
+        return NULL;
+    }
+
+    auto deserialize = plist::Format::Any::Deserialize(contents);
```

Comment:
The identified format wasn't being used later, so no need to compute it here: the second argument to `plist::Format::Any::Deserialize` is optional and detects the format automatically by default.